scripts/machine_translation/README.md

```diff
@@ -326,3 +326,17 @@ We used the same evaluation method (by averaging the last 10 checkpoints) as in
 ```
 BLEU+case.mixed+lang.en-de+numrefs.1+smooth.exp+test.wmt14/full+tok.13a+version.1.4.14 = 26.2 57.8/32.2/20.1/13.0 (BP = 0.994 ratio = 0.994 hyp_len = 62312 ref_len = 62688)
 ```
+#### Stochastic Beam Search
+
+Additionally, it has been shown that Stochastic Beam Search can obtain diverse translations with good quality in ["Stochastic Beams and Where to Find Them: The Gumbel-Top-k Trick for Sampling Sequences Without Replacement"](https://arxiv.org/pdf/1903.06059.pdf). To use stochactic Beam Search for evaluation, add the ``--stochastic`` option and set the ``--temperature``. The default temperature is 0.5. According to the paper, Lower temperature can yield better translations. There are some results for reference.
```

Comment:
You may show a complete example, for example:

```
SUBWORD_ALGO=yttm
python3 evaluate_transformer.py \
    --param_path ${SAVE_DIR}/avg_51_60.params \
    --src_lang en \
    --tgt_lang de \
    --cfg ${SAVE_DIR}/config.yml \
    --src_tokenizer ${SUBWORD_ALGO} \
    --tgt_tokenizer ${SUBWORD_ALGO} \
    --src_subword_model_path wmt2014_ende/${SUBWORD_ALGO}.model \
    --tgt_subword_model_path wmt2014_ende/${SUBWORD_ALGO}.model \
    --src_vocab_path wmt2014_ende/${SUBWORD_ALGO}.vocab \
    --tgt_vocab_path wmt2014_ende/${SUBWORD_ALGO}.vocab \
    --src_corpus wmt2014_ende/test.raw.en \
    --tgt_corpus wmt2014_ende/test.raw.de \
    --stochastic \
    --temperature 0.5 \
    --fp16
```

Reply:
You may use Transformer-base as an example and refer to the evaluation section of transformer-base by adding 

```
[Transformer Base](#transformer-base)
```

Reply:
Also, you may mention the reason why temperature = 1.0 will result in worse BLEU scores. According to the paper, this is because the exposure bias problem: "Because the model is not trained to use its own predictions as input, at test time errors can accumulate.  As a result, when sampling with the default temperatureT= 1, the expected BLEU score is very low (below 10). To improvequality of generated sentences we use lower temperaturesand experiment with T= 0.05, 0.1, 0.2, 0.5.  We then use different methods to estimate the BLEU score".